classes/class3/hdavidzhu_class3_homework/routes/cat-router.js

```diff
@@ -0,0 +1,101 @@
+var faker = require('faker');
+
+function shuffle(array) {
+  var currentIndex = array.length, temporaryValue, randomIndex ;
+
+  // While there remain elements to shuffle...
+  while (0 !== currentIndex) {
+
+    // Pick a remaining element...
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+
+    // And swap it with the current element.
+    temporaryValue = array[currentIndex];
+    array[currentIndex] = array[randomIndex];
+    array[randomIndex] = temporaryValue;
+  }
+
+  return array;
+}
+
+function getRandomColors() {
+	var colors = ['violet', 'coral', 'cerise', 'wheat', 'red', 'blue', 'green'];
+	shuffle(colors);
+
+	var chosenColorQueue = Math.ceil(Math.random() * 3);
+	return colors.slice(0, chosenColorQueue);
+}
+
+// Models.
+var Cat = require('../models/cat');
+module.exports = {
+	
+	newCat: function (req, res) {
+		var randomAge = Math.round(Math.random() * 14) + 1;
+		var randomName = faker.Name.firstName();
+		var randomColors = getRandomColors();
+
+		var cat = new Cat({
+			age: randomAge,
+			name: randomName,
+			colors: randomColors
+		});
+
+		cat.save(function (err) {
+		  if (err) {
+		    console.log("Problem saving bob", err);
+		  }
+		});
+
+		res.send("Time to make a new cat!");

```

Comment:
The response should be sent in the db callback to ensure that the save happened.